codegen/arrai/svc_types.arrai

```diff
@@ -69,6 +69,37 @@ let orderedTypes = \types
                         >> $`${go.name(.@item('name').s)} ${go.type(.@item('type'))}`
                     ::\i}
                 }
+
+                ${
+                let opParams = ep('restParams')?('queryParam')?.a:{} where $`${sysl.isPtr(.@item('type'))}`;
+                cond {opParams: $`
+                    func (r *${typename}) UnmarshalJSON(data []byte) error {
+                        inner := ${typename}{}
```

Comment:
```suggestion
                        var inner ${typename}
```

Reply:
json.Unmarshal will error out, if inner is nil.